Index: ceilometer/etc/ceilometer/ceilometer.conf.sample
===================================================================
--- ceilometer.orig/etc/ceilometer/ceilometer.conf.sample	2013-09-06 15:42:57.871578646 -0700
+++ ceilometer/etc/ceilometer/ceilometer.conf.sample	2013-09-06 15:42:57.863578646 -0700
@@ -111,7 +111,7 @@
 #
 
 # the filename to use with sqlite (string value)
-#sqlite_db=ceilometer.sqlite
+sqlite_db=sqlite:////var/lib/ceilometer/ceilometer.sqlite
 
 # If true, use synchronous mode for sqlite (boolean value)
 #sqlite_synchronous=true
