diff -Naurp ceilometer-2013.2.a175.gae9ccbd.orig/etc/ceilometer/ceilometer.conf.sample ceilometer-2013.2.a175.gae9ccbd/etc/ceilometer/ceilometer.conf.sample
--- ceilometer-2013.2.a175.gae9ccbd.orig/etc/ceilometer/ceilometer.conf.sample	2013-05-27 20:11:35.000000000 -0500
+++ ceilometer-2013.2.a175.gae9ccbd/etc/ceilometer/ceilometer.conf.sample	2013-05-27 20:16:48.569019601 -0500
@@ -497,6 +497,7 @@
 
 # Database connection string (string value)
 #database_connection=mongodb://localhost:27017/ceilometer
+database_connection=sqlite:////var/lib/ceilometer/ceilometer.sqlite
 
 
 #
