Index: ceilometer/etc/ceilometer/ceilometer.conf.sample
===================================================================
--- a/etc/ceilometer/ceilometer.conf.sample   2013-06-04 17:46:52.000000000 +0000
+++ b/etc/ceilometer/ceilometer.conf.sample        2013-06-05 09:36:45.074895860 +0000
@@ -517,6 +517,7 @@
 # The SQLAlchemy connection string used to connect to the
 # database (string value)
 #connection=sqlite:////common/db/$sqlite_db
+connection=sqlite:////var/lib/ceilometer/ceilometer.sqlite
 
 # timeout before idle sql connections are reaped (integer
 # value)
